MSAL/src/MSALAccount.m

```diff
@@ -161,7 +161,8 @@ - (BOOL)isEqual:(id)object
 - (NSUInteger)hash
 {
     NSUInteger hash = 0;
-    hash = hash * 31 + self.environment.hash;
+    // Equality of MSALAccount is depending on equality of homeAccountId or username
+    // So we are not able to calculate a precise hash
     return hash;
 }
```

Comment:
Should we remove the hash function here? Fix the hash value to 0 seems incorrect in general.

Reply:
`hash` is required if we want to define our own `isEqual` as indicated [here](https://developer.apple.com/documentation/objectivec/1418956-nsobject/1418795-isequal).
And it also requires [here](https://developer.apple.com/documentation/objectivec/1418956-nsobject/1418859-hash):
> If two objects are equal (as determined by the isEqual: method), they must have the same hash value.

So I think we may need to keep it anyway even though it doesn't help with performance :P

Reply:
Thank for the info, I see we implement the own logic in `isEqual` so we don't need the hash function other than for the procedure :D